includes/class-wc-subscription.php

```diff
@@ -634,22 +634,26 @@ public function is_manual() {
 	}
 
 	/**
-	 * Overrides the WC Order get_status function for draft and auto-draft statuses for a subscription
-	 * so that it will return a pending status instead of draft / auto-draft.
+	 * Sets the subscription status.
 	 *
-	 * @since 1.0.0 - Migrated from WooCommerce Subscriptions v2.0
-	 * @return string Status
+	 * Overrides the WC Order set_status() function to handle 'draft' and 'auto-draft' statuses for a subscription.
+	 *
+	 * 'draft' and 'auto-draft' statuses are WP statuses applied to the post when a subscription is created via admin. When
+	 * a subscription is being read from the database, and the status is set to the post's 'draft' or 'auto-draft' status, the
+	 * subscription status is treated as the default status - 'pending'.
+	 *
+	 * @since subscriptions-core 5.1.0
+	 *
+	 * @param string $new_status The new status.
+	 * @param string $note       Optional. The note to add to the subscription.
+	 * @param bool   $manual     Optional. Is the status change triggered manually? Default is false.
 	 */
-	public function get_status( $context = 'view' ) {
```

Comment:
In this PR i've deleted this function. 

The `WC_Subscription` override of `get_status()` only differed from the parent version by handling post 'draft' and 'auto-draft' statuses. Those are NOT core subscription statuses and only exist when you create a subscription using the admin window. WP creates the `shop_subscription` post with a draft status.  

To still handle this scenario, when you load a subscription, and the status is read from the database and set via `set_status()`, if at that point the subscription is being read as draft, it gets read in as pending.

I believe this change is functionally the same. A subscription's status in CRUD is only accessible via `get_status()` and `set_status()` and so this change only differs in internal data. Previously the internal subscription data had the status as 'draft' but would return 'pending'.  Now the internal data is 'pending' and returned as 'pending'.